File: listaExercicios/Escola/escola.py
class Professor:
    def __init__(self, nome):
        self.nome = nome
        self.escolas = []

    def adicionar_escola(self, escola):
        self.escolas.append(escola)

class Escola:
    def __init__(self, nome):
        self.nome = nome
        self.professores = []

    def adicionar_professor(self, professor):
        self.professores.append(professor)
        professor.adicionar_escola(self)  

escola1 = Escola("Colégio Alfa")
escola2 = Escola("Colégio Beta")

professor1 = Professor("João da Silva")
professor2 = Professor("Maria Souza")


escola1.adicionar_professor(professor1)
escola1.adicionar_professor(professor2)
escola2.adicionar_professor(professor1)


print(f"As escolas em que {professor1.nome} leciona são:")
for escola in professor1.escolas:
    print(escola.nome)

File: listaExercicios/Escola/Escola.java
import java.util.ArrayList;
import java.util.List;

class Professor {
    public String nome;
    public List<Escola> escolas = new ArrayList<>();

    public Professor(String nome) {
        this.nome = nome;
    }

    public void adicionarEscola(Escola escola) {
        escolas.add(escola);
    }
}

class Escola {
    public String nome;
    public List<Professor> professores = new ArrayList<>();

    public Escola(String nome) {
        this.nome = nome;
    }

    public void adicionarProfessor(Professor professor) {
        professores.add(professor);
        professor.adicionarEscola(this);
    }

    public static void main(String[] args) {
        Escola escola1 = new Escola("Colégio Alfa");
        Escola escola2 = new Escola("Colégio Beta");

        Professor professor1 = new Professor("João da Silva");
        Professor professor2 = new Professor("Maria Souza");

        escola1.adicionarProfessor(professor1);
        escola1.adicionarProfessor(professor2);
        escola2.adicionarProfessor(professor1);

        System.out.printf("As escolas em que %s leciona são:%n", professor1.nome);
        for (Escola escola : professor1.escolas) {
            System.out.println(escola.nome);
        }
    }
}